File: src/main/java/com/webflux/model/metadata/EnglishCourseMetadata.java
package com.webflux.model.metadata;

import lombok.*;
import lombok.experimental.SuperBuilder;

import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
@ToString(callSuper = true)
public class EnglishCourseMetadata extends CourseMetadata {

    private String level;
    private List<String> books;
}

File: src/main/java/com/webflux/model/metadata/EnglishCourseMetadata.kt
package com.webflux.model.metadata

import com.fasterxml.jackson.annotation.JsonTypeName

@JsonTypeName(CourseConstants.ENGLISH_COURSE_TYPE)
data class EnglishCourseMetadata(
    val level: String,
    val books: List<String>,
    override val type: String?
) : CourseMetadata(type)